lib/Zonemaster/GUI.pm

```diff
@@ -0,0 +1,136 @@
+package Zonemaster::GUI;
+
+use strict;
+use warnings;
+
+use Dancer2;
+use Carp;
+use Data::Dumper;
+use I18N::AcceptLanguage;
+use TOML::Tiny qw(from_toml);
+use File::Slurp qw( read_file );
+use File::ShareDir qw( dist_dir );
+
+
+our $VERSION = '0.0.0';
+
+# Generate configuration
+get '/:lang/app.config.json' => sub {
+    send_as JSON => config->{client_config}
+};
+
+# Serve additional assets
+get '/:lang/additional/**' => sub {
+    my ( $path ) = splat;
+
+    if ( grep /^\.{2,}$/, @{$path} ) {
```

Comment:
This might not be sufficient. What happens if I request `/fr/additional/bogus/../../../../../etc/passwd` or something like that?

Reply:
When using `**` splat return an array of path component, one component per element, so if I am not missing anything, in your example `$path` would be `( 'bogus', '..', '..', '..', '..', '..', 'etc', 'passwd')` and the grep would work.

[Relevant documentation as reference](https://metacpan.org/dist/Dancer2/view/lib/Dancer2/Manual.pod#Wildcard-Matching)

Reply:
Oh okay, my bad.